apps/charterafrica/src/components/Datasets/DatasetCard.js

```diff
@@ -0,0 +1,107 @@
+import { Box, Grid, Typography, Chip, Button } from "@mui/material";
+import { useState } from "react";
+
+import LineClampedRichTypography from "@/charterafrica/components/LineClampedRichTypography";
+import formatDateTime from "@/charterafrica/utils/formatDate";
+
+function DatasetCard({
+  formats,
+  notes,
+  title,
+  created,
+  readMore,
+  readLess,
+  updated,
+  author,
+  sx,
+}) {
+  const [showAll, setShowAll] = useState(false);
+
+  const handleClick = () => setShowAll(!showAll);
+
+  return (
+    <Box
+      sx={{
+        border: "none",
+        p: 2.5,
+        ...sx,
+      }}
+    >
+      <Grid
+        container
+        spacing={2}
+        justifyContent="space-between"
+        direction={{ xs: "column", md: "row" }}
+      >
+        <Grid item xs={12} md={4}>
+          <LineClampedRichTypography
+            color="neutral.dark"
+            lineClamp={1}
+            order={1}
+            variant="h5SmallSemiBold"
+            sx={{ mb: 1 }}
+          >
+            {title}
+          </LineClampedRichTypography>
+          <Typography variant="p1" color="neutral.main" sx={{ mb: 1 }}>
+            Updated {formatDateTime(updated, { includeTime: false })} | Created{" "}
+            {formatDateTime(created, { includeTime: false })}
+          </Typography>
+          <Typography variant="p1SemiBold" color="neutral.main">
+            {author}
+          </Typography>
+        </Grid>
+        <Grid item xs={12} md={4} order={{ xs: 4, sm: 3 }}>
+          <LineClampedRichTypography
+            color="common.black"
+            lineClamp={showAll ? -1 : 3}
+            order={1}
+            variant="p1"
+            sx={{ mb: 1 }}
+          >
+            {notes}
+          </LineClampedRichTypography>
+          <Button
+            sx={{
+              "&:hover": { backgroundColor: "transparent" },
+              color: "neutral.dark",
+              typography: "p1",
+              textDecoration: "underline",
+              padding: 0,
+            }}
+            onClick={handleClick}
+          >
+            {showAll ? readLess : readMore}
+          </Button>
+        </Grid>
+        <Grid
+          item
+          xs={12}
+          md={4}
+          order={{ xs: 3, sm: 4 }}
+          display="flex"
+          justifyContent={{ xs: "flex-start", sm: "flex-end" }}
+          alignItems="center"
+        >
+          {formats?.map((format) => (
```

Comment:
This whole `Grid` shouldn't be there if there are no `formats` to display, right?